addons/sale/models/account_move.py

```diff
@@ -74,11 +74,17 @@ def _reverse_moves(self, default_values_list=None, cancel=False):
     def action_post(self):
         #inherit of the function from account.move to validate a new tax and the priceunit of a downpayment
         res = super(AccountMove, self).action_post()
-        line_ids = self.mapped('line_ids').filtered(lambda line: any(line.sale_line_ids.mapped('is_downpayment')))
-        for line in line_ids:
+        downpayment_lines = self.line_ids.sale_line_ids.filtered('is_downpayment')
+        other_so_lines = downpayment_lines.order_id.order_line - downpayment_lines
+        real_invoices = other_so_lines.invoice_lines.move_id
```

Comment:
Reduce time complexity of the loop below
```suggestion
        real_invoices = set(other_so_lines.invoice_lines.move_id)
```